python/ray/dashboard/memory.py

```diff
@@ -0,0 +1,280 @@
+import base64
+import typing
+
+from collections import defaultdict
+from enum import Enum
+
+import ray
+
+
+def decode_object_id_if_needed(object_id: str) -> bytes:
+    """Decode objectID bytes string.
+
+    gRPC reply contains an objectID that is encodded by Base64.
+    This function is used to decode the objectID.
+    Note that there are times that objectID is already decoded as
+    a hex string. In this case, just convert it to a binary number.
+    """
+    if object_id.endswith("="):
+        # If the object id ends with =, that means it is base64 encoded.
+        # Object ids will always have = as a padding
+        # when it is base64 encoded because objectID is always 20B.
+        return base64.standard_b64decode(object_id)
+    else:
+        return ray.utils.hex_to_binary(object_id)
+
+
+class SortingType(Enum):
+    PID = 1
+    OBJECT_SIZE = 3
+    REFERENCE_TYPE = 4
+
+
+class GroupByType(Enum):
+    NODE_ADDRESS = 2
+
+
+class ReferenceType:
+    # We don't use enum because enum is not json serializable.
+    ACTOR_HANDLE = "ACTOR_HANDLE"
+    PINNED_IN_MEMORY = "PINNED_IN_MEMORY"
+    LOCAL_REFERENCE = "LOCAL_REFERENCE"
+    USED_BY_PENDING_TASK = "USED_BY_PENDING_TASK"
+    CAPTURED_IN_OBJECT = "CAPTURED_IN_OBJECT"
+    UNKNOWN_STATUS = "UNKNOWN_STATUS"
+
+
+class MemoryTableEntry:
+    def __init__(self, *, object_ref: dict, node_address: str, is_driver: bool,
```

Comment:
I think I understand why you're using the asterisks--to force key word args. Is that a convention in the Ray codebase? I haven't come across it that often in my python use.